docs/proposals/20200304-autoscaler-scale-ux.md

```diff
@@ -0,0 +1,117 @@
+
+We will need input from the autoscaler team, as well as input from api reviewers to ensure we obtain the best solution.
+
+These ideas require a particular (possibly new) cluster-api cloud provider controller
+which can inspect machinesets and annotate them as appropriate.  This could be
+done dynamically via billing API (not recommended) or have values compiled in
+statically.
+
+We could also attempt to build this logic into the machineset/machinedeployment
+controller, but this would probably not be very scalable and would require
+a well defined set of fields on the provider CRD.
+
+## Motivation
+
+### Goals
+
+### Non-Goals
+
+## Proposal
+
+### User Stories
+
+As a cluster admin, I want to add the ability to automatically scale my cluster
+using cluster autoscaler and the cluster-api.  In order to save money, I need
+the ability to scale from zero for a given machineset or machinedeployment.
+
+## Proposal Ideas
+
+There are several proposals to consider.  Once we decide on one, we'll eliminate
+the others.
+
+### Annotations
+Modify the autoscaler to look at particular annotations on a given object to determine the appropriate amounts of each resource.
+
+#### Benefits
+Easy, simple.  Users can modify if needed (we can instruct controllers to only
+add annotations if they're missing, etc), users can bring their own clouds
+and annotate their objects to support scale from zero and we don't need to worry
+about supporting every cloud day 1.
+
+#### Cons
+We're essentially creating an API around an annotation, and the api folks don't
+like that type of thing.
+
+### Spec Field
+We can add fields to the spec of a particular object to set these items.
+
+#### Benefits
+Well defined API.
+
+Users can set these fields, or some controller can
+populate them.
+
+#### Cons
+Automatically populating Spec fields is not ideal.  For starters, a user might
+copy an existing machineset to create a new one with a different instance type,
+and might overlook these fields because they didn't set them.  This would
+definitely cause problems.
+
+What do we do if the user specifies some fields and not others.
+
+Automatically populating spec seems to violate 'spec == user intent' principle.
+
+### Status Field
+We can add fields to the status of an object.
+
+#### Benefits
+Makes the most sense.  If it's information we're determining programatically
+about an object that other things need to consume, it should probably go here.
+
+Well defined API.
+
+#### Cons
+Users can't easily update this field if they want to set values themselves for
```

Comment:
This is just that kubectl makes it hard to update status?  Personally I'd rather tackle that (e.g. kubectl plugin) vs working around it with the override, if so!

Reply:
If it was just a matter of updating status, it'd be one thing. The other issue is that status is meant to be "recreatable" and if the resources are backed up/restored or migrated from one cluster to another the status will be lost.